src/advanced_inv.cpp

```diff
@@ -542,6 +542,58 @@ void advanced_inv_area::init()
             }
             break;
     }
+    if( veh != nullptr && vstor >= 0 ) {
+        const auto &part = veh->parts[vstor];
+        const auto label = veh->get_label( part.mount.x, part.mount.y );
+        if( !label.empty() ) {
+            desc = label;
+        }
+    }
+
+    // assemble a list of interesting traits of the target square
+
+    // fields? with a special case for fire
+    bool danger_field = false;
+    const field &tmpfld = g->m.field_at(x, y);
+    for( auto &fld : tmpfld ) {
+        const field_entry &cur = fld.second;
+        field_id curType = cur.getFieldType();
+        switch (curType) {
+            case fd_fire:
+                flags.append(_(" <color_white_red>FIRE</color>"));
+                break;
+            default:
+                if (cur.is_dangerous()) {
+                    danger_field = true;
+                }
+                break;
+        }
+    }
+    if (danger_field) {
+        flags.append(_(" <color_pink>DANGER</color>"));
+    }
+    
+    // trap?
+    const trap_id tid = g->m.tr_at(x, y);
+    if (tid != tr_null) {
+        const struct trap &t = *traplist[tid];
+        if ((t.can_see(g->u, x, y)) && !t.is_benign()) {
+            flags.append(_(" <color_ltgray>TRAP</color>"));
+        }
+    }
+
+    // water?
+    const ter_id ter = g->m.ter(x, y);
+    if ( (ter == t_water_dp || ter == t_water_pool || ter == t_swater_dp) ||
+         (ter == t_water_sh || ter == t_swater_sh || ter == t_sewage) ) {
+        flags.append(_(" <color_blue>WATER</color>"));
```

Comment:
I colored FIRE because it's especially important. I colored water because someone told me items dropped in deep water are destroyed, but I think I figured out that they were mistaken. So I can probably put water and fields back to the default color. I'll leave traps the default color too, for now, although some day I want to be able to set off traps by throwing/dropping items on them, in which case I'd want to color it here as high-importance.